python/pyspark/statcounter.py

```diff
@@ -53,12 +54,14 @@ def merge(self, value):
         return self
 
     # Merge another StatCounter into this one, adding up the internal statistics.
-    def mergeStats(self, other):
+    def mergeStats(self, other: "StatCounter") -> "StatCounter":
         if not isinstance(other, StatCounter):
             raise TypeError("Can only merge StatCounter but got %s" % type(other))
 
         if other is self:  # reference equality holds
-            self.merge(copy.deepcopy(other))  # Avoid overwriting fields in a weird order
+            self.merge(
+                cast(float, copy.deepcopy(other))
+            )  # Avoid overwriting fields in a weird order
```

Comment:
How can we cast `cast(float, copy.deepcopy(other))` here?
I guess this is a long-standing bug and should call `self.mergeStats` instead of `self.merge`? cc @HyukjinKwon